### PR TITLE
Set up Python workflow and Dependabot for pip and github-actions

### PR DESCRIPTION
**Pull request by mmorel-35**

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Lint with flake8
+      run: |
+        pip install flake8
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
```
Comment (fffonion) [inline .github/workflows/python-package.yml:43]: looks like we are silencing all lint outputs here, is that expected or should we fix the lint
as well to make this step more meaningful?

**Review by fffonion — COMMENTED** [review]
Thank you @mmorel-35 . I left a comment on the lint step.

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (mmorel-35) [reply .github/workflows/python-package.yml:43]: Hi @fffonion, I found this [here](https://docs.github.com/en/actions/guides/building-and-testing-python#starting-with-the-python-workflow-template)

**Comment by fffonion** [comment]
Thanks again @mmorel-35 ! I've merged the PR. Will take some time to fix lint and re-add lint in CI.

**Pull request merged**